### DISCUSSION TITLE
Unifiy backend tree structure

### DISCUSSION THREAD

**Discussion by heliocastro** [eclipse-sw360/sw360#2602] [Ideas]
Code-wise, SW360 backend is cluttered with multiple subfolders, separating the library modules from the tomcat servlet code, making maintenance complex.

The idea is to eliminate backend/src/src-<module>, backend/svc/svc-module and common properties and unify them in a single structure under backend.

An initial [layout branch exists already](https://github.com/eclipse-sw360/sw360/tree/feature/unify_backend) and all the base maven structure is rearranged to a cleaner view.

Some issues need to be resolved. Modules like projects and vulnerabilities are used as both servlets and as a library themselves. This will force a split between the servlet and the shared library code.

The opportunity to execute this refactor arises now since we are cleaning SW360 codebase.

**Comment by GMishx** [comment]
Since we do not have liferay in picture anymore, I also agree with unifying the libraries. What are your opinions on this @smrutis1 ?

**Comment by KoukiHama** [comment]
I fully support this refactor. Simplifying the backend structure will greatly improve maintainability and make future development more efficient.
